lib/versioncontrol/upgradewindow/upgradewindow.go

```diff
@@ -40,16 +40,25 @@ const (
 	// kubeSchedKey is the key under which the kube controller schedule is exported
 	kubeSchedKey = "agent-maintenance-schedule"
 
+	// kubeVersionKey is the key under which the kube controller version is exported
+	kubeVersionKey = "agent-auth-version"
+
 	// unitScheduleFile is the name of the file to which the unit schedule is exported.
 	unitScheduleFile = "schedule"
 
+	// unitVersionFile is the name of the file to which the version is exported.
+	unitVersionFile = "version"
```

Comment:
`version` feels way too generic, maybe `auth-version`?

```suggestion
	unitVersionFile = "auth-version"
```